qa/tasks/cephfs/test_misc.py

```diff
@@ -198,6 +198,16 @@ def test_dump_inode_hexademical(self):
         info = self.fs.mds_asok(['dump', 'inode', hex(ino)])
         assert info['path'] == "/foo"
 
+    def test_fs_lsflags(self):
+        self.fs.set_joinable(False)
+        self.fs.set_allow_new_snaps(False)
+        self.fs.set_allow_multimds_snaps(False)
+        self.fs.set_allow_standby_replay(True)
+        lsflags = json.loads(self.fs.mon_manager.raw_cluster_cmd('fs', 'lsflags', self.fs.name, "--format=json-pretty"))
+        self.assertEqual(lsflags["not_joinable"], True)
+        self.assertEqual(lsflags["allow_snaps"], False)
+        #self.assertEqual(lsflags["allow_multimds_snaps"], False)
```

Comment:
I'm not sure how to set `allow_multimds_snaps`. It doesn't work for me as mentioned in [MonCommands](https://github.com/ceph/ceph/blob/master/src/mon/MonCommands.h#L317). Any suggestions?

Reply:
What does the debug output show?

Reply:
I meant it cannot be set like [this](https://github.com/ceph/ceph/pull/39660/files#diff-501f5d864f91e98b1c214da58065308a41554b87ce6366f2314a84c19c287ca7R596). 

Gives the error:
```
2021-03-09 17:23:34,222.222 INFO:__main__:======================================================================
2021-03-09 17:23:34,222.222 INFO:__main__:ERROR: test_fs_lsflags (tasks.cephfs.test_misc.TestMisc)
2021-03-09 17:23:34,222.222 INFO:__main__:----------------------------------------------------------------------
2021-03-09 17:23:34,223.223 INFO:__main__:Traceback (most recent call last):
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/qa/tasks/cephfs/test_misc.py", line 204, in test_fs_lsflags
2021-03-09 17:23:34,223.223 INFO:__main__:    self.fs.set_allow_multimds_snaps(False)
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/qa/tasks/cephfs/filesystem.py", line 596, in set_allow_multimds_snaps
2021-03-09 17:23:34,223.223 INFO:__main__:    self.set_var("allow_multimds_snaps", yes)
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/qa/tasks/cephfs/filesystem.py", line 575, in set_var
2021-03-09 17:23:34,223.223 INFO:__main__:    self.mon_manager.raw_cluster_cmd("fs", "set", self.name, var, *a)
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/build/../qa/tasks/vstart_runner.py", line 1005, in raw_cluster_cmd
2021-03-09 17:23:34,223.223 INFO:__main__:    return self.run_cluster_cmd(**kwargs).stdout.getvalue()
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/build/../qa/tasks/vstart_runner.py", line 995, in run_cluster_cmd
2021-03-09 17:23:34,223.223 INFO:__main__:    return self.controller.run(**kwargs)
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/build/../qa/tasks/vstart_runner.py", line 399, in run
2021-03-09 17:23:34,223.223 INFO:__main__:    return self._do_run(**kwargs)
2021-03-09 17:23:34,223.223 INFO:__main__:  File "/home/jcollin/workspace/ceph/build/../qa/tasks/vstart_runner.py", line 467, in _do_run
2021-03-09 17:23:34,224.224 INFO:__main__:    proc.wait()
2021-03-09 17:23:34,224.224 INFO:__main__:  File "/home/jcollin/workspace/ceph/build/../qa/tasks/vstart_runner.py", line 213, in wait
2021-03-09 17:23:34,224.224 INFO:__main__:    raise CommandFailedError(self.args, self.exitstatus)
2021-03-09 17:23:34,224.224 INFO:__main__:teuthology.exceptions.CommandFailedError: Command failed with status 22: ['./bin/ceph', 'fs', 'set', 'cephfs', 'allow_multimds_snaps', 'false']
```

Reply:
There should be a message to `stderr`, what does it say?

Reply:
I think Let's test if the `lsflags` print some default flag values too. Please see the latest test code.